src/main/java/com/epam/aidial/core/log/GfLogStore.java

```diff
@@ -239,8 +238,9 @@ static String assembleStreamingResponse(Buffer response) {
             JsonNode systemFingerprint = null;
             JsonNode model = null;
             JsonNode choices = null;
-            // each chunk is separated by one or multiple new lines with the prefix: 'data:'
-            scanner.useDelimiter("\n*data: *");
+            // each chunk is separated by one or multiple new lines with the prefix: 'data:' (except the first chunk)
+            // chunks may contain `data:` inside chunk data, which may lead to incorrect parsing
+            scanner.useDelimiter("(^data: *|\n+data: *)");
```

Comment:
Please add the same fix to https://github.com/epam/ai-dial-core/blob/fix-assemble-streaming-response/src/main/java/com/epam/aidial/core/util/ModelCostCalculator.java#L87